tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,586 @@
+      </Window.Content>
+    </Window>
+  );
+};
+
+const TypeContent = (props) => {
+  return (
+    <Stack>
+      <Stack.Item width="14px" textAlign="center">
+        <Icon name={TYPE_ICONS[props.type]} />
+      </Stack.Item>
+      <Stack.Item grow style={{ 'text-transform': 'capitalize' }}>
+        {props.type.toLowerCase()}
+      </Stack.Item>
+      <Stack.Item>
+        {props.type === 'Can Make' ? (
+          props.craftable
+        ) : props.diet.liked_food.includes(props.type) ? (
+          <Icon name="face-laugh-beam" color={'good'} />
+        ) : props.diet.disliked_food.includes(props.type) ? (
+          <Icon name="face-tired" color={'average'} />
+        ) : (
+          props.diet.toxic_food.includes(props.type) && (
+            <Icon name="skull-crossbones" color={'bad'} />
+          )
+        )}
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const RecipeContentCompact = (props, context) => {
+  const { act } = useBackend<Data>(context);
+  const item = props.item;
+  return (
+    <Section my={0.5}>
+      <Stack my={-0.75}>
+        <Stack.Item>
+          <Box
+            className={
+              item.result?.includes('/datum/reagent/')
+                ? classes([
+                  'default_containers32x32',
+                  item.result?.replace(/[/_]/gi, ''),
+                ])
+                : classes(['food32x32', item.result?.replace(/[/_]/gi, '')])
+            }
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Stack>
+            <Stack.Item grow>
+              <Box bold mb={0.5} style={{ 'text-transform': 'capitalize' }}>
+                {item.name}
+              </Box>
+              <Box style={{ 'text-transform': 'capitalize' }} color={'gray'}>
+                {Array.from(
+                  item.reqs.map((i) =>
+                    i.is_reagent
+                      ? i.name + '\xa0' + i.amount + 'u'
+                      : i.amount > 1
+                        ? i.name + '\xa0' + i.amount + 'x'
+                        : i.name
+                  )
+                ).join(', ')}
+              </Box>
+            </Stack.Item>
+            <Stack.Item>
+              {!item.is_guide && (
+                <Button
+                  my={0.3}
+                  lineHeight={2.5}
+                  align="center"
+                  content="Make"
+                  disabled={!props.craftable || props.busy}
+                  icon={props.busy ? 'circle-notch' : 'utensils'}
+                  iconSpin={props.busy ? 1 : 0}
+                  onClick={() =>
+                    act('make', {
+                      recipe: item.ref,
+                    })
+                  }
+                />
+              )}
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+const RecipeContent = (props, context) => {
+  const { act } = useBackend<Data>(context);
+  const item: Recipe = props.item;
```

Comment:
we should be destructuring as before, but if you want to explicitly type item as a Recipe, either the props should be typed or item should be immediately destructured as a Recipe
```
({ item }: PropsType, context) or similar